src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -35,83 +53,150 @@ export interface DeleteWorkspaceHookArgs {
 }
 
 
-export const useDeleteWorkspaceState = (hookArgs: DeleteWorkspaceHookArgs) : DeleteWorkspaceState => {
+export const useDeleteWorkspaceState = (hookArgs: DeleteWorkspaceHookArgs): DeleteWorkspaceState => {
   const [deleting, setDeleting] = useState(false)
   const [loading, setLoading] = useState(false)
-  const [apps, setApps] = useState<DeleteWorkspaceModalLeoApp[]>()
   const [collaboratorEmails, setCollaboratorEmails] = useState<string[] | null>()
   const [workspaceBucketUsageInBytes, setWorkspaceBucketUsageInBytes] = useState<number | null>()
-  const [controlledResourcesExist, setControlledResourcesExist] = useState(false)
-  const [deletableApps, nonDeletableApps] = _.partition(isResourceDeletable('app'), apps) as [DeleteWorkspaceModalLeoApp[], DeleteWorkspaceModalLeoApp[]]
+  const [deletingResources, setDeletingResources] = useState(false)
+  const [workspaceResources, setWorkspaceResources] = useState<WorkspaceResources | null>(null)
 
   const workspaceInfo: WorkspaceInfo = hookArgs.workspace.workspace
   const signal = useCancellation()
+  const checkAzureResourcesTimeout = useRef<number>()
+
+  const fetchWorkspaceResources = async (workspace: BaseWorkspace): Promise<WorkspaceResources> => {
+    const apps = isGoogleWorkspace(workspace) ?
+        await Ajax(signal).Apps.listWithoutProject({
+          role: 'creator',
+          saturnWorkspaceName: workspaceInfo.name
+        }) as DeleteWorkspaceModalLeoApp[] :
+        await Ajax(signal).Apps.listAppsV2(workspaceInfo.workspaceId) as DeleteWorkspaceModalLeoApp[]
+
+    // only v2 runtimes supported right now for azure
```

Comment:
But what runtimes (v1?) in Google workspaces?

Reply:
If this starts returning runtimes for google workspaces, it will introduce new behavior that will show the "undeletable workspace warning" for runtimes in creating which is differing behavior from the current modal as shipped.  

Today's deletion modal allows google workspaces to proceed with deletion regardless of runtime status; I think this should be reevaluated as I don't think it's great ux, but I don't think it should be in scope for this PR.